src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3079,36 +3081,36 @@ async def do_create(self, app, data):
-            ('atime', None, str.lower),
-            ('casesensitivity', None, str.lower),
-            ('comments', 'org.freenas:description', None),
-            ('compression', None, str.lower),
-            ('copies', None, lambda x: str(x)),
-            ('deduplication', 'dedup', str.lower),
-            ('exec', None, str.lower),
-            ('managedby', 'org.truenas:managedby', None),
-            ('quota', None, _none),
-            ('quota_warning', 'org.freenas:quota_warning', str),
-            ('quota_critical', 'org.freenas:quota_critical', str),
-            ('readonly', None, str.lower),
-            ('recordsize', None, None),
-            ('refquota', None, _none),
-            ('refquota_warning', 'org.freenas:refquota_warning', str),
-            ('refquota_critical', 'org.freenas:refquota_critical', str),
-            ('refreservation', None, _none),
-            ('reservation', None, _none),
-            ('snapdir', None, str.lower),
-            ('sparse', None, None),
-            ('sync', None, str.lower),
-            ('volblocksize', None, None),
-            ('volsize', None, lambda x: str(x)),
-            ('xattr', None, str.lower),
-            ('special_small_block_size', 'special_small_blocks', None),
+        for i, real_name, transform, inheritable in (
+            ('aclmode', None, str.lower, False),
+            ('acltype', None, str.lower, False),
+            ('atime', None, str.lower, False),
+            ('casesensitivity', None, str.lower, False),
+            ('comments', 'org.freenas:description', None, False),
+            ('compression', None, str.lower, False),
+            ('copies', None, lambda x: str(x), False),
+            ('deduplication', 'dedup', str.lower, False),
+            ('exec', None, str.lower, False),
+            ('managedby', 'org.truenas:managedby', None, False),
+            ('quota', None, _none, False),
+            ('quota_warning', 'org.freenas:quota_warning', str, False),
+            ('quota_critical', 'org.freenas:quota_critical', str, False),
+            ('readonly', None, str.lower, False),
+            ('recordsize', None, None, False),
+            ('refquota', None, _none, False),
+            ('refquota_warning', 'org.freenas:refquota_warning', str, False),
+            ('refquota_critical', 'org.freenas:refquota_critical', str, False),
+            ('refreservation', None, _none, False),
+            ('reservation', None, _none, False),
+            ('snapdir', None, str.lower, False),
+            ('sparse', None, None, False),
+            ('sync', None, str.lower, False),
+            ('volblocksize', None, None, False),
+            ('volsize', None, lambda x: str(x), False),
+            ('xattr', None, str.lower, False),
+            ('special_small_block_size', 'special_small_blocks', None, True),
         ):
-            if i not in data:
+            if i not in data or (inheritable and data[i] == 'INHERIT'):
```

Comment:
Can't you just remove all inheritable with `INHERIT` value from `data` instead of introducing such a large diff?

How is `special_small_block_size` different from any other dataset properties in this case? Why do we allow it to be explicitly inherited while others are inherited when they are not specified?